graphql/types.go

```diff
@@ -173,6 +173,20 @@ type SelectionSet struct {
 	Fragments  []*Fragment
 }
 
+// ShallowCopy returns a shallow copy of SelectionSet.
+func (s *SelectionSet) ShallowCopy() *SelectionSet {
+	cp := &SelectionSet{}
+	if s.Selections != nil {
+		cp.Selections = make([]*Selection, len(s.Selections))
+		copy(cp.Selections, s.Selections)
+	}
+	if s.Fragments != nil {
+		cp.Fragments = make([]*Fragment, len(s.Fragments))
+		copy(cp.Fragments, s.Fragments)
+	}
+	return cp
+}
```

Comment:
Just in case: do you want to add a unit test? (What's the expectation for this package?)

Reply:
I added a unit test!